enterprise/backend/test/metabase_enterprise/sso/api/saml_test.clj

```diff
@@ -0,0 +1,21 @@
+(ns metabase-enterprise.sso.api.saml-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.test :as mt]))
+
+(deftest saml-settings-test
+  (testing "PUT /api/saml/settings"
+    (testing "Valid SAML settings can be saved via an API call"
+      (mt/user-http-request :crowberto :put 200 "saml/settings" {:saml-keystore-path "test_resources/keystore.jks"
+                                                                 :saml-keystore-password "123456"
+                                                                 :saml-keystore-alias "sp"})
+
+      (testing "Invalid SAML settings returns 400"
```

Comment:
The `testing` contexts don't need to be nested here (or below). It'll mean that if a test fails, all the contexts will be printed in the output like:
```
;; saml-settings-test: PUT /api/saml/settings Valid SAML settings can be saved via an API call Invalid SAML settings returns 400 Blank SAML settings returns 200 / PUT http://localhost:3000/api/saml/settings expected a status code of 200, got 404. (Line: 170)
expected: (= expected-status-code actual-status-code)
  actual: (not (= 200 404))
```